rust/src/dns/dns.rs

```diff
@@ -330,6 +362,8 @@ pub struct DNSState {
     pub request_buffer: Vec<u8>,
     pub response_buffer: Vec<u8>,
 
+    config: ConfigTracker,
```

Comment:
I think this looks good. Just wonder if it would make sense to make this an `Option<ConfigTracker>` as we won't be using it in the common case most likely.

Reply:
Make sense.  This change is done.